packages/wallet-sdk/src/provider/SubscriptionManager.ts

```diff
@@ -23,11 +23,11 @@ export interface SubscriptionNotification {
 
 export class SubscriptionManager {
   private readonly subscriptionMiddleware: SubscriptionMiddleware;
-  readonly events: SafeEventEmitter;
+  readonly events: EventEmitter;
 
-  constructor(provider: Web3Provider & SafeEventEmitter) {
+  constructor(provider: Web3Provider & EventEmitter) {
     const blockTracker = new PollingBlockTracker({
-      provider: provider as Provider,
+      provider: provider as unknown as Provider,
```

Comment:
why do we have `Web3Provider` AND `Provider` 😢

Reply:
Provider is the type that comes from the `eth-block-tracker` package, which provides the `PollingBlockTracker` class